src/h5web/providers/my-utils.ts

```diff
@@ -104,46 +132,55 @@ export function withMyInterpretation<
   ]);
 }
 
-export function makeMyNxDataGroup(
-  name: string,
-  signal: MyHDF5Dataset,
-  opts: Opts = {}
-): MyHDF5Group {
-  return makeMyGroup(name, [signal], {
-    ...opts,
-    attributes: [
-      ...(opts.attributes || []),
-      makeStrAttr('NX_class', 'NXdata'),
-      makeStrAttr('signal', signal.name),
-    ],
-  });
-}
-
-export function makeMyNxDataGroupWithAxes(
+export function makeMyNxDataGroup<
+  U extends Record<string, MyHDF5Dataset<HDF5SimpleShape, HDF5NumericType>>
+>(
   name: string,
   opts: {
     signal: MyHDF5Dataset<HDF5SimpleShape, HDF5NumericType>;
-    axes: Record<string, MyHDF5Dataset<HDF5SimpleShape, HDF5NumericType>>;
-    axesAttr: (keyof typeof opts.axes | '.')[];
-  } & Opts
+    errors?: MyHDF5Dataset<HDF5SimpleShape, HDF5NumericType>;
+    title?: MyHDF5Dataset<HDF5ScalarShape, HDF5StringType>;
+    silxStyle?: SilxStyle;
+  } & AllOrNone<{ axes: U; axesAttr: (keyof U | '.')[] }> &
+    GroupOpts
 ): MyHDF5Group {
-  const { signal, axes, axesAttr, ...groupOpts } = opts;
-
-  return makeMyGroup(name, [signal, ...Object.values(axes)], {
-    ...groupOpts,
-    attributes: [
-      ...(groupOpts.attributes || []),
-      makeStrAttr('NX_class', 'NXdata'),
-      makeStrAttr('signal', signal.name),
-      makeStrAttr('axes', axesAttr),
+  const {
+    signal,
+    title,
+    errors,
+    silxStyle,
+    axes = {},
+    axesAttr,
+    ...groupOpts
+  } = opts;
+
+  return makeMyGroup(
+    name,
+    [
+      signal,
+      ...(title ? [title] : []),
+      ...(errors ? [errors] : []),
```

Comment:
I wish there was an easier way to write this...

Reply:
Hehehe me too, I even looked it up 😄 -- either you wrap a `push` in an `if`, or you `filter` for `undefined` afterwards...